api/src/modules/countries/countries.service.ts

```diff
@@ -1,47 +1,45 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
 import { Country } from '@shared/dto/countries/country.entity';
 import { CreateCountryDto } from '@shared/dto/countries/create-country.dto';
 import { Repository } from 'typeorm';
 
 @Injectable()
 export class CountriesService {
-  constructor(private countryRepository: Repository<Country>) {}
-  private readonly country: Country[] = [];
+  constructor(@InjectRepository(Country)  private countryRepository: Repository<Country>) {}
 
-  create(country: CreateCountryDto) {
-    this.countryRepository.save(country);
+  async save(createCountryDto: CreateCountryDto) {
+    return this.countryRepository.save(createCountryDto);
   }
 
-  findAll(): Country[] {
-    return this.countryRepository.findAll();
+  async find(): Promise<Country[]> {
+    return this.countryRepository.find();
   }
 
-  findOne(id: number) {
-    return this.countryRepository.findOne((country) => country.id === id);
+  async findOneBy(id: string) {
+    return this.countryRepository.findOne({
+      where: { id },
+    });
   }
 
-  update(id: number) {
-    const index = this.countryRepository.findIndex(
-      (country) => country.id === id,
-    );
-    if (index === -1) {
-      return 'Country not found';
-    }
-    this.countryRepository[index] = {
-      ...this.countryRepository[index],
-      name: 'Country updated',
-    };
-    return 'Country updated';
+  async update(body, id: string) {
+    const countryToUpdate = await this.countryRepository.findOneBy({
+      id: id ,
+    });
+
+    return this.countryRepository.save(countryToUpdate)
```

Comment:
Uhm.. I don't think you are actually updating the country here. You are just saving the same country you just found in the database. I would say you are duplicating it 😬 .

Look for an `update` method or similar in the repository and use the `body` you are receiving.